learning_resources/etl/loaders.py

```diff
@@ -609,30 +614,35 @@ def load_podcast(podcast_data: dict) -> LearningResource:
         )
 
         episode_ids = []
-
-        for episode_data in episodes_data:
-            episode = load_podcast_episode(episode_data)
-            episode_ids.append(episode.id)
-
-        unpublished_episode_ids = (
-            learning_resource.children.filter(
-                relation_type=LearningResourceRelationTypes.PODCAST_EPISODES.value,
+        if learning_resource.published:
+            for episode_data in episodes_data:
+                episode = load_podcast_episode(episode_data)
+                episode_ids.append(episode.id)
+
+            unpublished_episode_ids = (
+                learning_resource.children.filter(
+                    relation_type=LearningResourceRelationTypes.PODCAST_EPISODES.value,
+                )
+                .exclude(child__id__in=episode_ids)
+                .values_list("child__id", flat=True)
             )
-            .exclude(child__id__in=episode_ids)
-            .values_list("child__id", flat=True)
-        )
-        LearningResource.objects.filter(id__in=unpublished_episode_ids).update(
-            published=False
-        )
-        episode_ids.extend(unpublished_episode_ids)
-        learning_resource.resources.set(
-            episode_ids,
-            through_defaults={
-                "relation_type": LearningResourceRelationTypes.PODCAST_EPISODES,
-            },
-        )
+            LearningResource.objects.filter(id__in=unpublished_episode_ids).update(
+                published=False
+            )
+            episode_ids.extend(unpublished_episode_ids)
+            learning_resource.resources.set(
+                episode_ids,
+                through_defaults={
+                    "relation_type": LearningResourceRelationTypes.PODCAST_EPISODES,
+                },
+            )
+
+    if not created and not learning_resource.published:
+        resource_unpublished_actions(learning_resource)
+    elif learning_resource.published:
+        resource_upserted_actions(learning_resource)
```

Comment:
Maybe these 4 lines of code could be put in a common function since they are called for podcasts, episodes, courses, and programs.